flow/project.py

```diff
@@ -2717,11 +3041,54 @@ def _verify_group_compatibility(self, groups):
         return all(a.isdisjoint(b) for a in groups for b in groups if a != b)
 
     def _verify_aggregate_project(self, aggregate):
-        """Verifies that all aggregates belongs to the same project."""
-        for job in aggregate:
-            if job not in self:
-                raise ValueError("Job {} is not present "
-                                 "in the project".format(job))
+        """Verifies that the aggregate belongs to the this project."""
+        return any(aggregate in stored_aggregator for stored_aggregator in self._stored_aggregators)
+
+    def _verify_aggregate_eligibility(self, aggregate, jobs):
+        """Verifies whether the aggregate is present in the jobs provided by the users
+        or not.
+        """
+        if jobs is None:
+            return True
+        return aggregate in jobs
+
+    def _get_aggregate_from_id(self, id):
+        # Iterate over all the instances of stored aggregates and search for the
+        # aggregate in those instances.
+        for stored_aggregator in self._stored_aggregators:
+            try:
+                return stored_aggregator[id]
+            except LookupError:  # Didn't found aggregate in this stored object
+                pass
```

Comment:
I would rather not use a try and except block here, checking `id in stored_aggregator` will likely be multiple times faster since we except failures here.

Reply:
the aggregate storing classes are implemented in such a way that if we use `__contains__` method then we'll have to pass an aggregate and not a id. If we use the `__get__` method then we'll be passing an id.
We could change the functionality if you suggest?